### PR TITLE
Change dependency warning link for PySocks

### PR DESCRIPTION
**Pull request by dslackw**

### CODE REVIEW

**Comment by pquentin** [comment]
Thanks! Did you mean to raise this against the 1.26 branch?

**Comment by dslackw** [comment]
Yes, if you think is the right branch.

File: src/urllib3/contrib/socks.py [hunk 1]
```diff
@@ -51,7 +51,7 @@
         (
             "SOCKS support in urllib3 requires the installation of optional "
             "dependencies: specifically, PySocks.  For more information, see "
-            "https://urllib3.readthedocs.io/en/latest/contrib.html#socks-proxies"
+            "https://urllib3.readthedocs.io/en/stable/advanced-usage.html#socks-proxies"
```
Comment (pquentin) [inline src/urllib3/contrib/socks.py:54]: I had missed from my phone that contrib was changed to advanced-usage, sorry. But this is the main branch, so we should stick with `latest` please:

```suggestion
            "https://urllib3.readthedocs.io/en/latest/advanced-usage.html#socks-proxies"
```

**Review by pquentin — CHANGES_REQUESTED** [review]
I had missed that you change

**Review by sethmlarson — APPROVED** [review]
LGTM, thank you!

**Review by illia-v — APPROVED** [review]

**Pull request merged**